multi/stm32l4-multi/flash.c

```diff
@@ -143,13 +143,13 @@ size_t flash_readData(uint32_t offset, char *buff, size_t size)
 }
 
 
-size_t flash_readOtp(uint32_t offset, char *buff, size_t size)
+ssize_t flash_readOtp(uint32_t offset, char *buff, size_t size)
 {
 	size_t ret;
 
 	ret = otp_isValidAddress(offset + OTP_ADDR, size);
 	if (!ret) {
```

Comment:
otp_isValidAddress has no side-effects, so it's ok for direct if usage. On the other hand !ret is against MISRA rule regarding casting int -> bool
```suggestion
	if (otp_isValidAddress(offset + OTP_ADDR, size) == 0) {
```